drivers/fpga/fpga_altera_agilex_bridge.c

```diff
@@ -0,0 +1,433 @@
+/*
+ * Copyright (c) 2024, Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT altr_socfpga_agilex_bridge
+
+#include <errno.h>
+#include <zephyr/device.h>
+#include <zephyr/sip_svc/sip_svc.h>
+#include <zephyr/drivers/sip_svc/sip_svc_agilex_smc.h>
+#include <zephyr/drivers/fpga.h>
+#include <zephyr/logging/log.h>
+#include "fpga_altera_agilex_bridge.h"
+
+LOG_MODULE_REGISTER(fpga_altera);
+
+static uint32_t mailbox_client_token;
+static struct sip_svc_controller *mailbox_smc_dev;
+
+#define MAX_TIMEOUT_MSECS (1 * 1000UL)
+
+static uint32_t mailbox_client_token;
+static struct sip_svc_controller *mailbox_smc_dev;
```

Comment:
The devicetree bindings for this driver should include a phandle property that refers to the sip_svc_controller.

That should be stored as compile-time const data in your `struct <driver>_config`.

Similarly, the mailbox client token should be part of a `struct <driver>_data` assuming it is mutable and only runtime discoverable.

Reply:
@cfriedt As discussed in the architecture review meeting, we have made a dummy node in the device tree for this driver as it doesn't need the information about the SiP SVC controller.  As the SiP SVC device node will be 
 returned by "sip_svc_get_controller" function declared in "zephyr/sip_svc/sip_svc.h" file and it is a subsystem, not a driver.

Yes, for the other comment regarding keeping the information in the struct <driver>_data, we agree it should be kept the way you mentioned.